automation/src/test/scala/org/broadinstitute/dsde/test/api/RawlsApiSpec.scala

```diff
@@ -351,5 +353,108 @@ class RawlsApiSpec extends TestKit(ActorSystem("MySpec")) with FreeSpecLike with
+          // service account should have owner access
+          assert(unconstrainedBucketAcl.exists(acl => acl.getRole().equals("OWNER") && acl.getEmail().endsWith(serviceAccountEmailDomain)))
+          val actualBucketRolesAndPolicies = unconstrainedBucketAcl.collect {
+            case acl if (acl.getRole() != null && !acl.getRole().equals("OWNER")) => (acl.getRole(), acl.getEmail())
+          }
+          val expectedBucketRolesAndPolicies = unconstrainedSamPolicies.collect {
+            case AccessPolicyResponseEntry("project-owner", AccessPolicyMembership(emails, _, _), _) => ("WRITER", emails.head)
+            case AccessPolicyResponseEntry(policyName, _, email) if policyToBucketAccessLevel.contains(policyName) => (policyToBucketAccessLevel(policyName), email.value)
+          }
+          actualBucketRolesAndPolicies should contain theSameElementsAs expectedBucketRolesAndPolicies
+
+          // check object acls
+          val unconstrainedObjectAcl = googleStorageDAO.getDefaultObjectAccessControls(bucketName).futureValue.getItems.asScala.toList
+          // service account should have owner access
+          assert(unconstrainedObjectAcl.exists(acl => acl.getRole().equals("OWNER") && acl.getEmail().endsWith(serviceAccountEmailDomain)))
+          val actualObjectRolesAndPolicies = unconstrainedObjectAcl.collect {
+            case acl if (acl.getRole() != null && !acl.getRole().equals("OWNER")) => (acl.getRole(), acl.getEmail())
+          }
+          val expectedObjectRolesAndPolicies = unconstrainedSamPolicies.collect {
+            case AccessPolicyResponseEntry("project-owner", AccessPolicyMembership(emails, _, _), _) => ("READER", emails.head)
+            case AccessPolicyResponseEntry(policyName, _, email) if policyToObjectAccessLevel.contains(policyName) => (policyToObjectAccessLevel(policyName), email.value)
+          }
+          actualObjectRolesAndPolicies should contain theSameElementsAs expectedObjectRolesAndPolicies
+        }
+      }
+    }
+
+    "should have correct policies in Sam and ACLs in Google when a constrained workspace is created" in {
+      implicit val patienceConfig: PatienceConfig = PatienceConfig(timeout = 20 seconds)
+      implicit val token: AuthToken = ownerAuthToken
+      val workspacePolicyNames = Set("can-compute", "project-owner", "writer", "reader", "share-writer", "can-catalog", "owner", "share-reader")
+      // bucket and object access levels for sam policies as described in comments in insertBucket function in HttpGoogleServicesDAO
+      val policyToBucketAccessLevel = Map("project-owner" -> "WRITER", "owner" -> "WRITER", "writer" -> "WRITER", "reader" -> "READER")
+      val policyToObjectAccessLevel = Map("project-owner" -> "READER", "owner" -> "READER", "writer" -> "READER", "reader" -> "READER")
+      val serviceAccountEmailDomain = "developer.gserviceaccount.com"
```

Comment:
Repeated blocks (workspacePolicyNames, policyToBucketAccessLevel, policyToObjectAccessLevel, serviceAccountEmailDomain) here can also be set once outside the tests and used in both places.